# **AdvDS - Assignment 1: Binary Classification - NBA Player Prediction - Data Preparation**

Build a binary classification model that can predict if a rookie player will last at least 5 years in the NBA

**Student Name:** Nathan Fragar

**Student No. :** 93087548

**Week:** 2

**Date:** 20FEB2022

**Team Name:** Group 1
* Nuwan Munasinghe
* Wenyingwuwy
* Nathan Fragar
* Sean Williams

## 1. Load and Discover Dataset

**[1.1]** Task: Import required packages: Pandas, Numpy, joblib

In [65]:
# Task: Import the pandas, numpy and joblib package
import pandas as pd
import numpy as np
import joblib as job

In [66]:
cd /home/jovyan/work

/home/jovyan/work


In [67]:
# Load custom functions
from src.indicator.sounds import beep_when_done

In [68]:
# Automatically reload modules
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


**[1.2]** Task: Identify correct directory

In [69]:
! ls -l data/raw

total 996
drwxrwxrwx 1 root root   4096 Feb 19 04:51 archive
-rwxrwxrwx 1 root root 322813 Feb  8 10:25 test.csv
-rwxrwxrwx 1 root root 695622 Feb  8 10:25 train.csv


**[1.3]** Task: Load Test and Train Files into Pandas Data Frames

In [70]:
# File URLs
file_path_train = "data/raw/train.csv"
file_path_test= "data/raw/test.csv"

# Load files into df_training and df_validation data frames
df_train = pd.read_csv(file_path_train)
df_test = pd.read_csv(file_path_test)

**[1.4]** Task: Display Train and Test Data Frames

In [71]:
# Task: Display first 5 rows of df_train
df_train.head()

Id  GP   MIN   PTS  FGM  FGA   FG%  3P Made  3PA   3P%  ...  FTA   FT%  \
0  10556  80  24.3   7.8  3.0  6.4  45.7      0.1  0.3  22.6  ...  2.9  72.1   
1   5342  75  21.8  10.5  4.2  7.9  55.1     -0.3 -1.0  34.9  ...  3.6  67.8   
2   5716  85  19.1   4.5  1.9  4.5  42.8      0.4  1.2  34.3  ...  0.6  75.7   
3  13790  63  19.1   8.2  3.5  6.7  52.5      0.3  0.8  23.7  ...  1.5  66.9   
4   5470  63  17.8   3.7  1.7  3.4  50.8      0.5  1.4  13.7  ...  0.5  54.0   

   OREB  DREB  REB  AST  STL  BLK  TOV  TARGET_5Yrs  
0   2.2   2.0  3.8  3.2  1.1  0.2  1.6            1  
1   3.6   3.7  6.6  0.7  0.5  0.6  1.4            1  
2   0.6   1.8  2.4  0.8  0.4  0.2  0.6            1  
3   0.8   2.0  3.0  1.8  0.4  0.1  1.9            1  
4   2.4   2.7  4.9  0.4  0.4  0.6  0.7            1  

[5 rows x 21 columns]

In [72]:
# Task: Display first 5 rows of df_train
df_test.head()

Id  GP   MIN   PTS  FGM   FGA   FG%  3P Made  3PA   3P%  FTM  FTA   FT%  \
0     1  56   9.1   4.0  1.6   3.7  43.7      0.1  0.3   7.3  0.7  1.2  63.4   
1  8194  43  19.3  10.1  3.7   8.1  46.0      0.6  1.7  35.1  1.8  2.5  75.3   
2     3  82  33.9  11.3  4.9  10.6  45.6      0.5  1.9  44.8  1.8  2.7  71.2   
3  8196  86  44.7  18.8  6.8  15.9  42.9      0.5  1.8  13.5  4.5  6.3  70.9   
4  8197  58  12.3   4.7  1.6   4.0  40.0      0.5  1.7  38.7  1.1  1.3  76.9   

   OREB  DREB  REB  AST  STL  BLK  TOV  
0   1.2   0.8  1.7  0.4  0.2  0.3  0.8  
1   0.5   0.9  1.5  3.5  0.6 -0.0  1.8  
2   1.3   3.3  4.5  2.5  1.3  0.3  2.0  
3   1.5   3.2  5.0  4.1  0.9  0.1  3.6  
4   0.2   0.6  0.9  1.5  0.5 -0.4  0.9

**Observation:** Both datasets are identical with the exception the Test Dataset does not have the Target Variable. This is to be expected

**[1.5]** Task: Display Train and Test Shape

In [73]:
# Task: Display shape of df_train
df_train.shape

(8000, 21)

In [74]:
# Task: Display shape of df_test
df_test.shape

(3799, 20)

**[1.6]** Task: Display descriptive statistics for Train and Test Datasets

In [75]:
# Task: Describe df_train
df_train.describe()

Id           GP          MIN          PTS          FGM  \
count   8000.000000  8000.000000  8000.000000  8000.000000  8000.000000   
mean    6856.971000    62.777875    18.576663     7.267087     2.807037   
std     3977.447579    17.118774     8.935263     4.318732     1.693373   
min        4.000000    -8.000000     2.900000     0.800000     0.300000   
25%     3413.750000    51.000000    12.000000     4.100000     1.600000   
50%     6787.500000    63.000000    16.800000     6.300000     2.400000   
75%    10299.250000    74.000000    23.500000     9.500000     3.700000   
max    13798.000000   123.000000    73.800000    34.200000    13.100000   

               FGA          FG%      3P Made          3PA          3P%  ...  \
count  8000.000000  8000.000000  8000.000000  8000.000000  8000.000000  ...   
mean      6.231213    44.608900     0.264525     0.816563    19.583700  ...   
std       3.584559     6.155453     0.384093     1.060964    16.003155  ...   
min       0.800000    21.300000    -1.100000    -3.100000   -38.500000  ...   
25%       3.600000    40.400000     0.000000     0.100000     8.400000  ...   
50%       5.400000    44.400000     0.300000     0.800000    19.500000  ...   
75%       8.100000    48.700000     0.500000     1.500000    30.600000  ...   
max      28.900000    67.200000     1.700000     4.700000    82.100000  ...   

               FTA          FT%         OREB         DREB          REB  \
count  8000.000000  8000.000000  8000.000000  8000.000000  8000.000000   
mean      1.947787    71.365825     1.077838     2.168500     3.245300   
std       1.252352    10.430447     0.785670     1.392224     2.085154   
min       0.000000   -13.300000     0.000000     0.200000     0.300000   
25%       1.000000    65.000000     0.500000     1.100000     1.700000   
50%       1.700000    71.400000     0.900000     1.900000     2.800000   
75%       2.600000    77.500000     1.500000     2.900000     4.300000   
max      11.100000   168.900000     5.500000    11.000000    15.900000   

               AST          STL          BLK          TOV  TARGET_5Yrs  
count  8000.000000  8000.000000  8000.000000  8000.000000  8000.000000  
mean      1.624513     0.648688     0.245212     1.257762     0.833625  
std       1.355986     0.407626     0.821037     0.723270     0.372440  
min       0.000000     0.000000   -17.900000     0.100000     0.000000  
25%       0.700000     0.300000     0.100000     0.700000     1.000000  
50%       1.300000     0.600000     0.200000     1.100000     1.000000  
75%       2.200000     0.900000     0.400000     1.600000     1.000000  
max      12.800000     3.600000    18.900000     5.300000     1.000000  

[8 rows x 21 columns]

**Observations**

**Outlier - Minimum Values** - Negative for GP, 3P Made, 3PA, FTM, BLK
* GP: -8 games
* 3P Made: -1.1 
* 3PA: -3.1
* FTM: 
* BLK: -17.9

**Outlier - Maximum Values**
* MIN - A max value of 73.8 minutes played per game appears high, particularly with 75% at 23.5 - may have some outlier records
* a NBA Game has 48 minutes per game


In [76]:
# Task: Describe df_train
df_test.describe()

Id           GP          MIN          PTS          FGM  \
count   3799.000000  3799.000000  3799.000000  3799.000000  3799.000000   
mean    7010.614109    62.853909    18.650224     7.328034     2.835404   
std     3954.173641    17.151740     8.727259     4.294724     1.688427   
min        1.000000     6.000000     3.700000     0.700000     0.300000   
25%     3644.000000    51.000000    12.200000     4.200000     1.600000   
50%     7062.000000    63.000000    17.000000     6.400000     2.500000   
75%    10402.500000    74.000000    23.300000     9.400000     3.700000   
max    13792.000000   126.000000    68.000000    33.000000    13.400000   

               FGA          FG%      3P Made          3PA          3P%  \
count  3799.000000  3799.000000  3799.000000  3799.000000  3799.000000   
mean      6.302580    44.599079     0.255962     0.796920    19.234746   
std       3.579221     6.040168     0.380987     1.052862    15.968989   
min       0.800000    25.100000    -1.000000    -2.700000   -38.000000   
25%       3.700000    40.500000     0.000000     0.100000     8.500000   
50%       5.500000    44.600000     0.300000     0.800000    19.400000   
75%       8.100000    48.500000     0.500000     1.500000    30.250000   
max      26.200000    74.600000     1.600000     4.300000    73.800000   

               FTM          FTA          FT%         OREB         DREB  \
count  3799.000000  3799.000000  3799.000000  3799.000000  3799.000000   
mean      1.399842     1.953567    71.612924     1.096025     2.179495   
std       0.926140     1.250376    10.457336     0.785678     1.371935   
min       0.000000     0.000000    23.700000     0.000000     0.200000   
25%       0.700000     1.000000    65.000000     0.500000     1.200000   
50%       1.200000     1.700000    71.500000     0.900000     1.900000   
75%       1.900000     2.600000    78.000000     1.500000     2.900000   
max       7.800000     9.800000   127.100000     6.900000    12.000000   

               REB          AST          STL          BLK          TOV  
count  3799.000000  3799.000000  3799.000000  3799.000000  3799.000000  
mean      3.275783     1.636483     0.653593     0.257726     1.257910  
std       2.070646     1.335496     0.410573     0.639660     0.712449  
min       0.300000     0.000000     0.000000    -7.100000     0.100000  
25%       1.800000     0.600000     0.400000     0.100000     0.700000  
50%       2.800000     1.300000     0.600000     0.200000     1.100000  
75%       4.300000     2.300000     0.900000     0.400000     1.600000  
max      18.500000     9.000000     2.700000    14.800000     5.200000

**Obervations**
* Negative Values exist in a number of columns
* Mean and Standard Deviation between train and test appear to be reasonably well aligned


**[1.7]** Task: Identify Missing Values

In [77]:
# Task: Display Summary (Info)
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 21 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Id           8000 non-null   int64  
 1   GP           8000 non-null   int64  
 2   MIN          8000 non-null   float64
 3   PTS          8000 non-null   float64
 4   FGM          8000 non-null   float64
 5   FGA          8000 non-null   float64
 6   FG%          8000 non-null   float64
 7   3P Made      8000 non-null   float64
 8   3PA          8000 non-null   float64
 9   3P%          8000 non-null   float64
 10  FTM          8000 non-null   float64
 11  FTA          8000 non-null   float64
 12  FT%          8000 non-null   float64
 13  OREB         8000 non-null   float64
 14  DREB         8000 non-null   float64
 15  REB          8000 non-null   float64
 16  AST          8000 non-null   float64
 17  STL          8000 non-null   float64
 18  BLK          8000 non-null   float64
 19  TOV   

**Oberservation** 
There are no missing values in this dataset

**[1.8]** Task: Investigate Column Details - Find Min Values for each columns

In [78]:
df_train.min()

Id              4.0
GP             -8.0
MIN             2.9
PTS             0.8
FGM             0.3
FGA             0.8
FG%            21.3
3P Made        -1.1
3PA            -3.1
3P%           -38.5
FTM             0.0
FTA             0.0
FT%           -13.3
OREB            0.0
DREB            0.2
REB             0.3
AST             0.0
STL             0.0
BLK           -17.9
TOV             0.1
TARGET_5Yrs     0.0
dtype: float64

**[1.8.1]** Task: Find Rows where GP (Games Played) is negative

**Observated Statistics**
The following columns do not appear to have reliable information, they do not fit the profile of the statistic
* 3P Made - Negative Numbers should not be possible
* 3PA - Points Attempted should always be the same or greater than Points Made
* 3P% - Calculation of % should be (Points Made / Points Attempted) within a certain tolerance of error (due to rounding)
* FTM - Negative Numbers should not be possible
* FTA - Points Attempted should always be the same or greater than Points Made
* FT% 
* BLK  - Negative Numbers should not be possible


**[1.9]** Task: Check for duplicates

In [79]:
# Identify Rows that are Duplicated
dup = df_train.duplicated()
df_train[dup]

Empty DataFrame
Columns: [Id, GP, MIN, PTS, FGM, FGA, FG%, 3P Made, 3PA, 3P%, FTM, FTA, FT%, OREB, DREB, REB, AST, STL, BLK, TOV, TARGET_5Yrs]
Index: []

[0 rows x 21 columns]

In [80]:
# Identify Rows that are Duplicated
dup = df_test.duplicated()
df_test[dup]

Empty DataFrame
Columns: [Id, GP, MIN, PTS, FGM, FGA, FG%, 3P Made, 3PA, 3P%, FTM, FTA, FT%, OREB, DREB, REB, AST, STL, BLK, TOV]
Index: []

## 2. Feature Engineering

No creation of new features at this time.

**[2.1]** Create copy of Training and Test Datasets


In [81]:
# Create copy of df_train and df_test
df_train_cleaned = df_train
df_test_cleaned = df_test

**[2.2]** Split into X & y

In [82]:
from src.data.sets import pop_target

In [83]:
X_train_cleaned, y_train_cleaned = pop_target(df = df_train_cleaned,target_col = 'TARGET_5Yrs',to_numpy=False)

In [84]:
X_test_cleaned = df_test_cleaned

**[2.2]** Task: Clear any negative values

In [85]:
from src.data.clean import null_numeric_values

In [86]:
null_numeric_values(X_train_cleaned, ['GP','3P Made','3PA','3P%','FT%','BLK'], '<', 0.1 )

Id    GP   MIN   PTS  FGM  FGA   FG%  3P Made  3PA   3P%  FTM  FTA  \
0     10556  80.0  24.3   7.8  3.0  6.4  45.7      0.1  0.3  22.6  2.0  2.9   
1      5342  75.0  21.8  10.5  4.2  7.9  55.1      0.4  1.2  34.9  2.4  3.6   
2      5716  85.0  19.1   4.5  1.9  4.5  42.8      0.4  1.2  34.3  0.4  0.6   
3     13790  63.0  19.1   8.2  3.5  6.7  52.5      0.3  0.8  23.7  0.9  1.5   
4      5470  63.0  17.8   3.7  1.7  3.4  50.8      0.5  1.4  13.7  0.2  0.5   
...     ...   ...   ...   ...  ...  ...   ...      ...  ...   ...  ...  ...   
7995   2996  32.0   9.2   1.8  0.7  1.8  40.3      0.4  1.2  23.1  0.4  0.6   
7996  11679  54.0   6.0   1.8  0.7  1.4  48.7      0.1  0.1   3.1  0.2  0.4   
7997   5537  85.0  28.2  10.7  4.0  9.0  45.1      0.2  0.6  23.6  2.8  3.9   
7998   1292  39.0   7.7   2.5  1.0  2.3  40.1      0.4  1.2  13.3  0.6  0.7   
7999   2900  49.0  19.2   4.8  1.7  5.1  32.6      0.7  2.4  41.3  0.8  1.3   

       FT%  OREB  DREB  REB  AST  STL  BLK  TOV  
0     72.1   2.2   2.0  3.8  3.2  1.1  0.2  1.6  
1     67.8   3.6   3.7  6.6  0.7  0.5  0.6  1.4  
2     75.7   0.6   1.8  2.4  0.8  0.4  0.2  0.6  
3     66.9   0.8   2.0  3.0  1.8  0.4  0.1  1.9  
4     54.0   2.4   2.7  4.9  0.4  0.4  0.6  0.7  
...    ...   ...   ...  ...  ...  ...  ...  ...  
7995  65.7   0.3   1.8  1.9  0.5  0.3  0.2  0.4  
7996  70.1   1.0   1.1  2.0  0.1  0.0  0.3  0.3  
7997  69.7   1.0   2.1  3.1  3.4  1.2  0.2  1.8  
7998  74.3   0.4   0.6  0.9  0.2  0.3  0.3  0.5  
7999  70.1   0.1   1.1  1.2  3.5  0.9  0.4  1.4  

[8000 rows x 20 columns]

In [87]:
null_numeric_values(X_test_cleaned, ['3P Made','3PA','3P%','BLK'], '<', 0.1 )

Id  GP   MIN   PTS  FGM   FGA   FG%  3P Made  3PA   3P%  FTM  FTA  \
0        1  56   9.1   4.0  1.6   3.7  43.7      0.1  0.3   7.3  0.7  1.2   
1     8194  43  19.3  10.1  3.7   8.1  46.0      0.6  1.7  35.1  1.8  2.5   
2        3  82  33.9  11.3  4.9  10.6  45.6      0.5  1.9  44.8  1.8  2.7   
3     8196  86  44.7  18.8  6.8  15.9  42.9      0.5  1.8  13.5  4.5  6.3   
4     8197  58  12.3   4.7  1.6   4.0  40.0      0.5  1.7  38.7  1.1  1.3   
...    ...  ..   ...   ...  ...   ...   ...      ...  ...   ...  ...  ...   
3794  8175  84  21.2   8.7  3.4   6.7  50.2      0.4  1.2  22.8  1.7  2.5   
3795  8176  49  16.3   6.4  2.9   6.6  44.4      0.4  1.2  19.8  1.0  1.9   
3796  8178  53   9.9   2.1  0.8   1.8  43.1      0.4  1.2   1.9  0.6  1.0   
3797  8181  89  38.3  14.5  5.4  11.8  45.2      0.5  1.2  24.7  2.5  2.9   
3798  8183  55  12.0   3.0  1.1   2.5  42.3      0.1  0.6  11.7  0.9  1.1   

       FT%  OREB  DREB  REB  AST  STL  BLK  TOV  
0     63.4   1.2   0.8  1.7  0.4  0.2  0.3  0.8  
1     75.3   0.5   0.9  1.5  3.5  0.6  0.4  1.8  
2     71.2   1.3   3.3  4.5  2.5  1.3  0.3  2.0  
3     70.9   1.5   3.2  5.0  4.1  0.9  0.1  3.6  
4     76.9   0.2   0.6  0.9  1.5  0.5  0.4  0.9  
...    ...   ...   ...  ...  ...  ...  ...  ...  
3794  68.1   1.9   2.3  3.9  1.5  0.6  0.3  2.0  
3795  50.2   1.7   2.8  4.4  0.4  0.4  0.4  0.7  
3796  63.9   0.7   1.0  1.7  0.4  0.4  0.2  0.5  
3797  89.2   1.5   4.0  5.5  3.7  1.3  0.3  2.4  
3798  76.7   0.7   2.0  2.6  0.6  0.3  0.2  1.2  

[3799 rows x 20 columns]

In [88]:
null_numeric_values(X_train_cleaned, ['3P%','FT%','FG%'], '>', 99.9 )

Id    GP   MIN   PTS  FGM  FGA   FG%  3P Made  3PA   3P%  FTM  FTA  \
0     10556  80.0  24.3   7.8  3.0  6.4  45.7      0.1  0.3  22.6  2.0  2.9   
1      5342  75.0  21.8  10.5  4.2  7.9  55.1      0.4  1.2  34.9  2.4  3.6   
2      5716  85.0  19.1   4.5  1.9  4.5  42.8      0.4  1.2  34.3  0.4  0.6   
3     13790  63.0  19.1   8.2  3.5  6.7  52.5      0.3  0.8  23.7  0.9  1.5   
4      5470  63.0  17.8   3.7  1.7  3.4  50.8      0.5  1.4  13.7  0.2  0.5   
...     ...   ...   ...   ...  ...  ...   ...      ...  ...   ...  ...  ...   
7995   2996  32.0   9.2   1.8  0.7  1.8  40.3      0.4  1.2  23.1  0.4  0.6   
7996  11679  54.0   6.0   1.8  0.7  1.4  48.7      0.1  0.1   3.1  0.2  0.4   
7997   5537  85.0  28.2  10.7  4.0  9.0  45.1      0.2  0.6  23.6  2.8  3.9   
7998   1292  39.0   7.7   2.5  1.0  2.3  40.1      0.4  1.2  13.3  0.6  0.7   
7999   2900  49.0  19.2   4.8  1.7  5.1  32.6      0.7  2.4  41.3  0.8  1.3   

       FT%  OREB  DREB  REB  AST  STL  BLK  TOV  
0     72.1   2.2   2.0  3.8  3.2  1.1  0.2  1.6  
1     67.8   3.6   3.7  6.6  0.7  0.5  0.6  1.4  
2     75.7   0.6   1.8  2.4  0.8  0.4  0.2  0.6  
3     66.9   0.8   2.0  3.0  1.8  0.4  0.1  1.9  
4     54.0   2.4   2.7  4.9  0.4  0.4  0.6  0.7  
...    ...   ...   ...  ...  ...  ...  ...  ...  
7995  65.7   0.3   1.8  1.9  0.5  0.3  0.2  0.4  
7996  70.1   1.0   1.1  2.0  0.1  0.0  0.3  0.3  
7997  69.7   1.0   2.1  3.1  3.4  1.2  0.2  1.8  
7998  74.3   0.4   0.6  0.9  0.2  0.3  0.3  0.5  
7999  70.1   0.1   1.1  1.2  3.5  0.9  0.4  1.4  

[8000 rows x 20 columns]

In [89]:
null_numeric_values(X_test_cleaned, ['3P%','FT%','FG%'], '>', 99.9 )

Id  GP   MIN   PTS  FGM   FGA   FG%  3P Made  3PA   3P%  FTM  FTA  \
0        1  56   9.1   4.0  1.6   3.7  43.7      0.1  0.3   7.3  0.7  1.2   
1     8194  43  19.3  10.1  3.7   8.1  46.0      0.6  1.7  35.1  1.8  2.5   
2        3  82  33.9  11.3  4.9  10.6  45.6      0.5  1.9  44.8  1.8  2.7   
3     8196  86  44.7  18.8  6.8  15.9  42.9      0.5  1.8  13.5  4.5  6.3   
4     8197  58  12.3   4.7  1.6   4.0  40.0      0.5  1.7  38.7  1.1  1.3   
...    ...  ..   ...   ...  ...   ...   ...      ...  ...   ...  ...  ...   
3794  8175  84  21.2   8.7  3.4   6.7  50.2      0.4  1.2  22.8  1.7  2.5   
3795  8176  49  16.3   6.4  2.9   6.6  44.4      0.4  1.2  19.8  1.0  1.9   
3796  8178  53   9.9   2.1  0.8   1.8  43.1      0.4  1.2   1.9  0.6  1.0   
3797  8181  89  38.3  14.5  5.4  11.8  45.2      0.5  1.2  24.7  2.5  2.9   
3798  8183  55  12.0   3.0  1.1   2.5  42.3      0.1  0.6  11.7  0.9  1.1   

       FT%  OREB  DREB  REB  AST  STL  BLK  TOV  
0     63.4   1.2   0.8  1.7  0.4  0.2  0.3  0.8  
1     75.3   0.5   0.9  1.5  3.5  0.6  0.4  1.8  
2     71.2   1.3   3.3  4.5  2.5  1.3  0.3  2.0  
3     70.9   1.5   3.2  5.0  4.1  0.9  0.1  3.6  
4     76.9   0.2   0.6  0.9  1.5  0.5  0.4  0.9  
...    ...   ...   ...  ...  ...  ...  ...  ...  
3794  68.1   1.9   2.3  3.9  1.5  0.6  0.3  2.0  
3795  50.2   1.7   2.8  4.4  0.4  0.4  0.4  0.7  
3796  63.9   0.7   1.0  1.7  0.4  0.4  0.2  0.5  
3797  89.2   1.5   4.0  5.5  3.7  1.3  0.3  2.4  
3798  76.7   0.7   2.0  2.6  0.6  0.3  0.2  1.2  

[3799 rows x 20 columns]

Task: Display df_train_cleaned Information

In [90]:
# Display df_train_cleaned information
X_train_cleaned.max()

Id         13798.0
GP           123.0
MIN           73.8
PTS           34.2
FGM           13.1
FGA           28.9
FG%           67.2
3P Made        1.7
3PA            4.7
3P%           82.1
FTM            8.1
FTA           11.1
FT%           99.9
OREB           5.5
DREB          11.0
REB           15.9
AST           12.8
STL            3.6
BLK           18.9
TOV            5.3
dtype: float64

In [91]:
X_test_cleaned.max()

Id         13792.0
GP           126.0
MIN           68.0
PTS           33.0
FGM           13.4
FGA           26.2
FG%           74.6
3P Made        1.6
3PA            4.3
3P%           73.8
FTM            7.8
FTA            9.8
FT%           99.9
OREB           6.9
DREB          12.0
REB           18.5
AST            9.0
STL            2.7
BLK           14.8
TOV            5.2
dtype: float64

**[222]** Task: Clean and recalculate % fields

In [92]:
from src.features.percentages import calculate_percentage

In [93]:
X_train_cleaned = calculate_percentage(X_train_cleaned,numerator_col='FGM',denominator_col='FGA',result_col='FG%',round_result=1)
X_train_cleaned = calculate_percentage(X_train_cleaned,numerator_col='3P Made',denominator_col='3PA',result_col='3P%',round_result=1)
X_train_cleaned = calculate_percentage(X_train_cleaned,numerator_col='FTM',denominator_col='FTA',result_col='FT%',round_result=1)

In [94]:
X_test_cleaned = calculate_percentage(X_test_cleaned,numerator_col='FGM',denominator_col='FGA',result_col='FG%',round_result=1)
X_test_cleaned = calculate_percentage(X_test_cleaned,numerator_col='3P Made',denominator_col='3PA',result_col='3P%',round_result=1)
X_test_cleaned = calculate_percentage(X_test_cleaned,numerator_col='FTM',denominator_col='FTA',result_col='FT%',round_result=1)

**[223]** Task: Create Features: FG_Missed, 3P_Missed, FT_Missed

In [95]:
X_train_cleaned['FG_Missed'] = X_train_cleaned['FGA'] - X_train_cleaned['FGM']
X_train_cleaned['3P_Missed'] = X_train_cleaned['3PA'] - X_train_cleaned['3P Made']
X_train_cleaned['FT_Missed'] = X_train_cleaned['FTA'] - X_train_cleaned['FTM']

In [96]:
X_test_cleaned['FG_Missed'] = X_test_cleaned['FGA'] - X_test_cleaned['FGM']
X_test_cleaned['3P_Missed'] = X_test_cleaned['3PA'] - X_test_cleaned['3P Made']
X_test_cleaned['FT_Missed'] = X_test_cleaned['FTA'] - X_test_cleaned['FTM']

In [97]:
X_train_cleaned['FG_Missed'] = X_train_cleaned['FG_Missed'].round(1)
X_train_cleaned['3P_Missed'] = X_train_cleaned['3P_Missed'].round(1)
X_train_cleaned['FT_Missed'] = X_train_cleaned['FT_Missed'].round(1)

In [98]:
X_test_cleaned['FG_Missed'] = X_test_cleaned['FG_Missed'].round(1)
X_test_cleaned['3P_Missed'] = X_test_cleaned['3P_Missed'].round(1)
X_test_cleaned['FT_Missed'] = X_test_cleaned['FT_Missed'].round(1)

In [99]:
X_train_cleaned.sort_values(by=['BLK'], ascending=True).head(10)

Id    GP   MIN   PTS  FGM  FGA   FG%  3P Made  3PA   3P%  ...  OREB  \
2114   3377  60.0  21.8  11.1  4.6  9.9  46.5      0.4  1.2  33.3  ...   0.7   
1688   3680  47.0  10.9   2.5  1.0  2.2  45.5      0.4  1.2  33.3  ...   0.4   
7018  10452  85.0  12.0   4.8  1.7  3.9  43.6      0.2  0.6  33.3  ...   0.7   
7017   8008  76.0  26.9   9.5  3.6  8.7  41.4      0.2  0.7  28.6  ...   0.5   
6406  11281  36.0  16.8   6.8  2.6  6.3  41.3      0.1  0.2  50.0  ...   0.8   
3736  10434  89.0  16.6   6.1  2.7  6.0  45.0      0.4  1.2  33.3  ...   0.9   
936   11627  57.0  16.2   7.4  2.6  4.7  55.3      0.1  0.1   7.9  ...   1.5   
1693   4605  96.0  23.9  10.1  4.0  8.3  48.2      0.4  1.2  33.3  ...   1.4   
5807   1336  76.0  32.2  10.1  3.2  8.8  36.4      1.2  3.6  33.3  ...   0.6   
934   11792  66.0  20.0   8.1  3.1  6.5  47.7      0.4  1.2  33.3  ...   1.0   

      DREB  REB  AST  STL  BLK  TOV  FG_Missed  3P_Missed  FT_Missed  
2114   1.3  2.0  2.7  0.8  0.1  1.0        5.3        0.8        0.0  
1688   1.0  1.2  2.0  0.4  0.1  1.0        1.2        0.8        0.3  
7018   1.9  2.4  1.9  0.3  0.1  1.5        2.2        0.4        0.5  
7017   1.4  1.9  5.1  1.5  0.1  1.5        5.1        0.5        0.8  
6406   1.3  1.9  1.8  1.1  0.1  1.5        3.7        0.1        0.4  
3736   1.0  1.8  1.1  0.7  0.1  1.2        3.3        0.8        0.4  
936    2.1  3.4  0.8  0.6  0.1  0.9        2.1        0.0        1.4  
1693   1.8  3.4  5.0  1.5  0.1  1.6        4.3        0.8        0.8  
5807   3.9  4.4  3.5  1.4  0.1  1.4        5.6        2.4        0.4  
934    2.9  3.8  2.6  0.4  0.1  2.1        3.4        0.8        0.9  

[10 rows x 23 columns]

In [100]:
X_train_cleaned.min()

Id            4.0
GP            1.0
MIN           2.9
PTS           0.8
FGM           0.3
FGA           0.8
FG%          25.0
3P Made       0.1
3PA           0.1
3P%           0.3
FTM           0.0
FTA           0.0
FT%           0.0
OREB          0.0
DREB          0.2
REB           0.3
AST           0.0
STL           0.0
BLK           0.1
TOV           0.1
FG_Missed     0.5
3P_Missed     0.0
FT_Missed     0.0
dtype: float64

In [101]:
X_test_cleaned.max()

Id           13792.0
GP             126.0
MIN             68.0
PTS             33.0
FGM             13.4
FGA             26.2
FG%             81.2
3P Made          1.6
3PA              4.3
3P%             80.0
FTM              7.8
FTA              9.8
FT%             99.8
OREB             6.9
DREB            12.0
REB             18.5
AST              9.0
STL              2.7
BLK             14.8
TOV              5.2
FG_Missed       13.6
3P_Missed        2.9
FT_Missed        3.2
dtype: float64

**[2.]** Task: Create Player Efficiency Rating

In [102]:
from src.features.player_efficiency_rating import calculate_player_efficiency_rating

In [103]:
calculate_player_efficiency_rating(X_train_cleaned)

Id    GP   MIN   PTS  FGM  FGA   FG%  3P Made  3PA   3P%  ...  AST  \
0     10556  80.0  24.3   7.8  3.0  6.4  46.9      0.1  0.3  33.3  ...  3.2   
1      5342  75.0  21.8  10.5  4.2  7.9  53.2      0.4  1.2  33.3  ...  0.7   
2      5716  85.0  19.1   4.5  1.9  4.5  42.2      0.4  1.2  33.3  ...  0.8   
3     13790  63.0  19.1   8.2  3.5  6.7  52.2      0.3  0.8  37.5  ...  1.8   
4      5470  63.0  17.8   3.7  1.7  3.4  50.0      0.5  1.4  35.7  ...  0.4   
...     ...   ...   ...   ...  ...  ...   ...      ...  ...   ...  ...  ...   
7995   2996  32.0   9.2   1.8  0.7  1.8  38.9      0.4  1.2  33.3  ...  0.5   
7996  11679  54.0   6.0   1.8  0.7  1.4  50.0      0.1  0.1   3.1  ...  0.1   
7997   5537  85.0  28.2  10.7  4.0  9.0  44.4      0.2  0.6  33.3  ...  3.4   
7998   1292  39.0   7.7   2.5  1.0  2.3  43.5      0.4  1.2  33.3  ...  0.2   
7999   2900  49.0  19.2   4.8  1.7  5.1  33.3      0.7  2.4  29.2  ...  3.5   

      STL  BLK  TOV  FG_Missed  3P_Missed  FT_Missed  PPER  NPER   PER  
0     1.1  0.2  1.6        3.4        0.2        0.9  26.8  11.8  15.0  
1     0.5  0.6  1.4        3.7        0.8        1.2  35.1  13.5  21.6  
2     0.4  0.2  0.6        2.6        0.8        0.2  16.2   9.8   6.4  
3     0.4  0.1  1.9        3.2        0.5        0.6  26.5  15.3  11.2  
4     0.4  0.6  0.7        1.7        0.9        0.3  21.0   7.8  13.2  
...   ...  ...  ...        ...        ...        ...   ...   ...   ...  
7995  0.3  0.2  0.4        1.1        0.8        0.2  19.5   9.4  10.1  
7996  0.0  0.3  0.3        0.7        0.0        0.2  24.2   9.6  14.6  
7997  1.2  0.2  1.8        5.0        0.4        1.1  26.4  14.0  12.4  
7998  0.3  0.3  0.5        1.3        0.8        0.1  25.2  13.5  11.7  
7999  0.9  0.4  1.4        3.4        1.7        0.5  22.2  14.4   7.8  

[8000 rows x 26 columns]

In [104]:
calculate_player_efficiency_rating(X_test_cleaned)

Id  GP   MIN   PTS  FGM   FGA   FG%  3P Made  3PA   3P%  ...  AST  \
0        1  56   9.1   4.0  1.6   3.7  43.2      0.1  0.3  33.3  ...  0.4   
1     8194  43  19.3  10.1  3.7   8.1  45.7      0.6  1.7  35.3  ...  3.5   
2        3  82  33.9  11.3  4.9  10.6  46.2      0.5  1.9  26.3  ...  2.5   
3     8196  86  44.7  18.8  6.8  15.9  42.8      0.5  1.8  27.8  ...  4.1   
4     8197  58  12.3   4.7  1.6   4.0  40.0      0.5  1.7  29.4  ...  1.5   
...    ...  ..   ...   ...  ...   ...   ...      ...  ...   ...  ...  ...   
3794  8175  84  21.2   8.7  3.4   6.7  50.7      0.4  1.2  33.3  ...  1.5   
3795  8176  49  16.3   6.4  2.9   6.6  43.9      0.4  1.2  33.3  ...  0.4   
3796  8178  53   9.9   2.1  0.8   1.8  44.4      0.4  1.2  33.3  ...  0.4   
3797  8181  89  38.3  14.5  5.4  11.8  45.8      0.5  1.2  41.7  ...  3.7   
3798  8183  55  12.0   3.0  1.1   2.5  44.0      0.1  0.6  16.7  ...  0.6   

      STL  BLK  TOV  FG_Missed  3P_Missed  FT_Missed  PPER  NPER   PER  
0     0.2  0.3  0.8        2.1        0.2        0.5  29.7  18.4  11.3  
1     0.6  0.4  1.8        4.4        1.1        0.7  32.9  18.5  14.4  
2     1.3  0.3  2.0        5.7        1.4        0.9  23.6  13.1  10.5  
3     0.9  0.1  3.6        9.1        1.3        1.8  25.1  16.4   8.7  
4     0.5  0.4  0.9        2.4        1.2        0.2  26.5  15.5  11.0  
...   ...  ...  ...        ...        ...        ...   ...   ...   ...  
3794  0.6  0.3  2.0        3.3        0.8        0.8  28.2  14.3  13.9  
3795  0.4  0.4  0.7        3.7        0.8        0.9  29.2  15.8  13.4  
3796  0.4  0.2  0.5        1.0        0.8        0.4  20.5   8.7  11.8  
3797  1.3  0.3  2.4        6.4        0.7        0.4  24.4  13.3  11.1  
3798  0.3  0.2  1.2        1.4        0.5        0.2  20.3  12.3   8.0  

[3799 rows x 26 columns]

In [105]:
X_train_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 26 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Id         8000 non-null   int64  
 1   GP         8000 non-null   float64
 2   MIN        8000 non-null   float64
 3   PTS        8000 non-null   float64
 4   FGM        8000 non-null   float64
 5   FGA        8000 non-null   float64
 6   FG%        8000 non-null   float64
 7   3P Made    8000 non-null   float64
 8   3PA        8000 non-null   float64
 9   3P%        8000 non-null   float64
 10  FTM        8000 non-null   float64
 11  FTA        8000 non-null   float64
 12  FT%        8000 non-null   float64
 13  OREB       8000 non-null   float64
 14  DREB       8000 non-null   float64
 15  REB        8000 non-null   float64
 16  AST        8000 non-null   float64
 17  STL        8000 non-null   float64
 18  BLK        8000 non-null   float64
 19  TOV        8000 non-null   float64
 20  FG_Misse

**[222]** Task: Save as Interim - train & test

In [106]:
df_train_interim = pd.concat([X_train_cleaned, y_train_cleaned], axis=1)

In [107]:
# Save to CSV - train & test.csv
df_train_interim.to_csv("data/interim/train.csv", index = False)
X_test_cleaned.to_csv("data/interim/test.csv", index = False)

**[2.3]** Task: Import and Normalise all statistic columns

In [108]:
# Import MinMaxScaler
from sklearn.preprocessing import MinMaxScaler
# Instantiate Scaler
scaler = MinMaxScaler()

In [109]:
X_train_cleaned_scaled_cols = X_train_cleaned.loc[:,X_train_cleaned.columns!='Id']
X_test_cleaned_scaled_cols =  X_test_cleaned.loc[:,X_test_cleaned.columns!='Id']

In [110]:
X_train_cleaned_scaled_cols.max()

GP           123.0
MIN           73.8
PTS           34.2
FGM           13.1
FGA           28.9
FG%           72.4
3P Made        1.7
3PA            4.7
3P%           80.0
FTM            8.1
FTA           11.1
FT%           99.5
OREB           5.5
DREB          11.0
REB           15.9
AST           12.8
STL            3.6
BLK           18.9
TOV            5.3
FG_Missed     17.3
3P_Missed      3.1
FT_Missed      3.1
PPER          58.9
NPER          37.3
PER           48.9
dtype: float64

In [111]:
X_test_cleaned_scaled_cols.max()

GP           126.0
MIN           68.0
PTS           33.0
FGM           13.4
FGA           26.2
FG%           81.2
3P Made        1.6
3PA            4.3
3P%           80.0
FTM            7.8
FTA            9.8
FT%           99.8
OREB           6.9
DREB          12.0
REB           18.5
AST            9.0
STL            2.7
BLK           14.8
TOV            5.2
FG_Missed     13.6
3P_Missed      2.9
FT_Missed      3.2
PPER          53.1
NPER          36.5
PER           40.0
dtype: float64

In [112]:
# Scale Columns
X_train_cleaned_scaled = scaler.fit_transform(X_train_cleaned_scaled_cols)
X_test_cleaned_scaled = scaler.fit_transform(X_test_cleaned_scaled_cols)

In [113]:
# Convert numpy arrany df_train_cleaned_scaled values back to df_train_cleaned_scaled data frame 
X_train_cleaned_scaled = pd.DataFrame(np.squeeze(X_train_cleaned_scaled), columns=X_train_cleaned_scaled_cols.columns) 
X_test_cleaned_scaled = pd.DataFrame(np.squeeze(X_test_cleaned_scaled), columns=X_test_cleaned_scaled_cols.columns) 

In [114]:
X_train_cleaned.columns

Index(['Id', 'GP', 'MIN', 'PTS', 'FGM', 'FGA', 'FG%', '3P Made', '3PA', '3P%',
       'FTM', 'FTA', 'FT%', 'OREB', 'DREB', 'REB', 'AST', 'STL', 'BLK', 'TOV',
       'FG_Missed', '3P_Missed', 'FT_Missed', 'PPER', 'NPER', 'PER'],
      dtype='object')

In [115]:
# Drop original columns with standard values
X_train_cleaned = X_train_cleaned.drop(X_train_cleaned_scaled.columns,axis=1)
X_test_cleaned = X_test_cleaned.drop(X_test_cleaned_scaled.columns,axis=1)

In [116]:
# Concatenate frames to re-construct with scaled values + target (as appropriate)
df_train_cleaned = pd.concat([X_train_cleaned, X_train_cleaned_scaled,y_train_cleaned], axis=1)
df_test_cleaned = pd.concat([X_test_cleaned, X_test_cleaned_scaled], axis=1)

In [117]:
df_train_cleaned.head(10)

Id        GP       MIN       PTS       FGM       FGA       FG%  3P Made  \
0  10556  0.647541  0.301834  0.209581  0.210938  0.199288  0.462025   0.0000   
1   5342  0.606557  0.266573  0.290419  0.304688  0.252669  0.594937   0.1875   
2   5716  0.688525  0.228491  0.110778  0.125000  0.131673  0.362869   0.1875   
3  13790  0.508197  0.228491  0.221557  0.250000  0.209964  0.573840   0.1250   
4   5470  0.508197  0.210155  0.086826  0.109375  0.092527  0.527426   0.2500   
5   1961  0.713115  0.241185  0.239521  0.265625  0.281139  0.369198   0.1875   
6   2677  0.565574  0.249647  0.185629  0.226562  0.163701  0.723629   0.0000   
7  11722  0.459016  0.207334  0.137725  0.109375  0.138790  0.236287   0.5000   
8   6148  0.368852  0.332863  0.254491  0.218750  0.227758  0.381857   0.3125   
9   3040  0.516393  0.430183  0.482036  0.476562  0.373665  0.666667   0.1250   

        3PA       3P%  ...       STL       BLK       TOV  FG_Missed  \
0  0.043478  0.414053  ...  0.305556  0.005319  0.288462   0.172619   
1  0.239130  0.414053  ...  0.138889  0.026596  0.250000   0.190476   
2  0.239130  0.414053  ...  0.111111  0.005319  0.096154   0.125000   
3  0.152174  0.466750  ...  0.111111  0.000000  0.346154   0.160714   
4  0.282609  0.444166  ...  0.111111  0.026596  0.115385   0.071429   
5  0.239130  0.414053  ...  0.111111  0.010638  0.192308   0.267857   
6  0.239130  0.100376  ...  0.111111  0.031915  0.211538   0.101190   
7  0.586957  0.398996  ...  0.166667  0.026596  0.038462   0.148810   
8  0.347826  0.439147  ...  0.194444  0.026596  0.307692   0.214286   
9  0.217391  0.338770  ...  0.111111  0.127660  0.442308   0.261905   

   3P_Missed  FT_Missed      PPER      NPER       PER  TARGET_5Yrs  
0   0.064516   0.290323  0.286667  0.227273  0.368715            1  
1   0.258065   0.387097  0.471111  0.278788  0.491620            1  
2   0.258065   0.064516  0.051111  0.166667  0.208566            1  
3   0.161290   0.193548  0.280000  0.333333  0.297952            1  
4   0.290323   0.096774  0.157778  0.106061  0.335196            1  
5   0.258065   0.258065  0.404444  0.409091  0.355680            1  
6   0.354839   0.225806  0.300000  0.157576  0.422719            1  
7   0.612903   0.129032  0.175556  0.203030  0.290503            1  
8   0.354839   0.096774  0.200000  0.251515  0.281192            1  
9   0.258065   0.387097  0.446667  0.251515  0.487896            1  

[10 rows x 27 columns]

In [118]:
df_test_cleaned.head(10)

Id        GP       MIN       PTS       FGM       FGA       FG%   3P Made  \
0     1  0.416667  0.083981  0.102167  0.099237  0.114173  0.312839  0.000000   
1  8194  0.308333  0.242613  0.291022  0.259542  0.287402  0.358047  0.333333   
2     3  0.633333  0.469673  0.328173  0.351145  0.385827  0.367089  0.266667   
3  8196  0.666667  0.637636  0.560372  0.496183  0.594488  0.305606  0.266667   
4  8197  0.433333  0.133748  0.123839  0.099237  0.125984  0.254973  0.266667   
5     6  0.441667  0.192846  0.198142  0.190840  0.145669  0.656420  0.200000   
6  8198  0.325000  0.071540  0.077399  0.083969  0.078740  0.435805  0.200000   
7  8202  0.391667  0.130638  0.123839  0.114504  0.129921  0.325497  0.000000   
8    14  0.291667  0.278383  0.222910  0.213740  0.204724  0.466546  0.133333   
9  8207  0.158333  0.097978  0.058824  0.053435  0.059055  0.318264  0.066667   

        3PA       3P%  ...       AST       STL       BLK       TOV  FG_Missed  \
0  0.047619  0.414787  ...  0.044444  0.074074  0.013605  0.137255   0.122137   
1  0.380952  0.439850  ...  0.388889  0.222222  0.020408  0.333333   0.297710   
2  0.428571  0.327068  ...  0.277778  0.481481  0.013605  0.372549   0.396947   
3  0.404762  0.345865  ...  0.455556  0.333333  0.000000  0.686275   0.656489   
4  0.380952  0.365915  ...  0.166667  0.185185  0.020408  0.156863   0.145038   
5  0.261905  0.414787  ...  0.066667  0.222222  0.020408  0.156863   0.091603   
6  0.261905  0.414787  ...  0.022222  0.037037  0.006803  0.058824   0.068702   
7  0.023810  0.624060  ...  0.088889  0.148148  0.020408  0.098039   0.137405   
8  0.190476  0.414787  ...  0.155556  0.185185  0.020408  0.254902   0.183206   
9  0.142857  0.355890  ...  0.044444  0.074074  0.006803  0.078431   0.061069   

   3P_Missed  FT_Missed      PPER      NPER     PER  
0   0.068966    0.15625  0.396907  0.441358  0.2825  
1   0.379310    0.21875  0.479381  0.444444  0.3600  
2   0.482759    0.28125  0.239691  0.277778  0.2625  
3   0.448276    0.56250  0.278351  0.379630  0.2175  
4   0.413793    0.06250  0.314433  0.351852  0.2750  
5   0.275862    0.31250  0.461340  0.160494  0.5725  
6   0.275862    0.06250  0.288660  0.262346  0.3225  
7   0.034483    0.03125  0.283505  0.302469  0.2850  
8   0.206897    0.12500  0.335052  0.228395  0.3950  
9   0.172414    0.03125  0.108247  0.194444  0.2025  

[10 rows x 26 columns]

**[222]** Task: Save files as processed

In [119]:
# Save to CSV - train & test.csv
df_train_cleaned.to_csv("data/processed/train.csv", index = False)
df_test_cleaned.to_csv("data/processed/test.csv", index = False)

**[2.4]** Task: Create Train and Validate Datasets

In [120]:
# Drop Target and ID for X, and create target from y
df_train_id = df_train_cleaned['Id']
X = df_train_cleaned.drop(['Id', 'TARGET_5Yrs'], axis=1)
y = df_train_cleaned['TARGET_5Yrs']

In [121]:
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 25 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   GP         8000 non-null   float64
 1   MIN        8000 non-null   float64
 2   PTS        8000 non-null   float64
 3   FGM        8000 non-null   float64
 4   FGA        8000 non-null   float64
 5   FG%        8000 non-null   float64
 6   3P Made    8000 non-null   float64
 7   3PA        8000 non-null   float64
 8   3P%        8000 non-null   float64
 9   FTM        8000 non-null   float64
 10  FTA        8000 non-null   float64
 11  FT%        8000 non-null   float64
 12  OREB       8000 non-null   float64
 13  DREB       8000 non-null   float64
 14  REB        8000 non-null   float64
 15  AST        8000 non-null   float64
 16  STL        8000 non-null   float64
 17  BLK        8000 non-null   float64
 18  TOV        8000 non-null   float64
 19  FG_Missed  8000 non-null   float64
 20  3P_Misse

In [122]:
from sklearn.model_selection import train_test_split

In [123]:
# split the data in training and remaining dataset. 80:10:10 for train:valid:test
X_train, X_val, y_train, y_val  = train_test_split(X, y, train_size=0.8, shuffle = True, stratify=y, random_state = 8)

**[2.5]** Task: Save Datasets: X_train, X_val, y_train, y_val

In [124]:
from src.data.sets import save_sets

In [125]:
save_sets(X_train=X_train, y_train=y_train, X_val=X_val, y_val=y_val, X_test=None, y_test=None, path='data/processed/')

**[2.5]** Task: Check train and validate datasets have been distributed well

In [126]:
y.value_counts(normalize=True)

1    0.833625
0    0.166375
Name: TARGET_5Yrs, dtype: float64

In [127]:
y_train.value_counts(normalize=True)

1    0.833594
0    0.166406
Name: TARGET_5Yrs, dtype: float64